# STAT 301: Individual Assignment 1

**Group:** 26

**Group member:** Sam Thorne 83910448

## New York City Airbnb Data:

This data originally retrieved from [Kaggle](https://www.kaggle.com/datasets/dgomonov/new-york-city-airbnb-open-data). The raw data can additionally be found on [GitHub](https://raw.githubusercontent.com/samnthorne/FinalProject/main/data/AB_NYC_2019.csv).

The link is: https://www.kaggle.com/datasets/dgomonov/new-york-city-airbnb-open-data

This dataset is a summary of Airbnb information in the New York City area during 2019. The data includes review information, host metrics, general metrics about the Bnb, and booking information. This is a public dataset that can be found on [insideairbnb.com].(http://insideairbnb.com/)

### Sample of Airbnb Data:

In [2]:
library(tidyverse)

In [3]:
# Reading in the data set:

data <- read_csv('https://raw.githubusercontent.com/samnthorne/FinalProject/main/data/AB_NYC_2019.csv')

head(data)

Rows: 48895 Columns: 16
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (5): name, host_name, neighbourhood_group, neighbourhood, room_type
dbl  (10): id, host_id, latitude, longitude, price, minimum_nights, number_o...
date  (1): last_review

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


id   name                                             host_id host_name  
1 2539 Clean & quiet apt home by the park               2787    John       
2 2595 Skylit Midtown Castle                            2845    Jennifer   
3 3647 THE VILLAGE OF HARLEM....NEW YORK !              4632    Elisabeth  
4 3831 Cozy Entire Floor of Brownstone                  4869    LisaRoxanne
5 5022 Entire Apt: Spacious Studio/Loft by central park 7192    Laura      
6 5099 Large Cozy 1 BR Apartment In Midtown East        7322    Chris      
  neighbourhood_group neighbourhood latitude longitude room_type       price
1 Brooklyn            Kensington    40.64749 -73.97237 Private room    149  
2 Manhattan           Midtown       40.75362 -73.98377 Entire home/apt 225  
3 Manhattan           Harlem        40.80902 -73.94190 Private room    150  
4 Brooklyn            Clinton Hill  40.68514 -73.95976 Entire home/apt  89  
5 Manhattan           East Harlem   40.79851 -73.94399 Entire home/apt  80  
6 Manhattan           Murray Hill   40.74767 -73.97500 Entire home/apt 200  
  minimum_nights number_of_reviews last_review reviews_per_month
1  1               9               2018-10-19  0.21             
2  1              45               2019-05-21  0.38             
3  3               0               <NA>          NA             
4  1             270               2019-07-05  4.64             
5 10               9               2018-11-19  0.10             
6  3              74               2019-06-22  0.59             
  calculated_host_listings_count availability_365
1 6                              365             
2 2                              355             
3 1                              365             
4 1                              194             
5 1                                0             
6 1                              129

In [4]:
summary(data)

       id               name              host_id           host_name        
 Min.   :    2539   Length:48895       Min.   :     2438   Length:48895      
 1st Qu.: 9471945   Class :character   1st Qu.:  7822033   Class :character  
 Median :19677284   Mode  :character   Median : 30793816   Mode  :character  
 Mean   :19017143                      Mean   : 67620011                     
 3rd Qu.:29152178                      3rd Qu.:107434423                     
 Max.   :36487245                      Max.   :274321313                     
                                                                             
 neighbourhood_group neighbourhood         latitude       longitude     
 Length:48895        Length:48895       Min.   :40.50   Min.   :-74.24  
 Class :character    Class :character   1st Qu.:40.69   1st Qu.:-73.98  
 Mode  :character    Mode  :character   Median :40.72   Median :-73.96  
                                        Mean   :40.73   Mean   :-73.95  
           

In [39]:
print(paste("There are", length(unique(data$name)), "Airbnbs"))
print(paste("There are", length(unique(data$host_id)), "different host ids"))
print(paste("There are", length(unique(data$host_name)), "host names"))
print(paste("There are", length(unique(data$neighbourhood_group)), "different neighbourhood groups"))
print(paste("There are", length(unique(data$neighbourhood)), "different neighbourhoods"))
print(paste("There are", length(unique(data$room_type)), "different room types"))

[1] "There are 47895 Airbnbs"
[1] "There are 37457 different host ids"
[1] "There are 11453 host names"
[1] "There are 5 different neighbourhood groups"
[1] "There are 221 different neighbourhoods"
[1] "There are 3 different room types"


In [40]:
# viewing unique values for some nominal attributes

# 221 unique values of neighbourhood. All have capital words
# print(unique(data$neighbourhood))

# 5 unique values for neighborhood group
print(unique(data$neighbourhood_group))

# 3 unique values for room type
print(unique(data$room_type))

[1] "Brooklyn"      "Manhattan"     "Queens"        "Staten Island"
[5] "Bronx"        
[1] "Private room"    "Entire home/apt" "Shared room"    


---

### Descriptive Summary:

**New York City Airbnb Dataset** has 16 attributes describing various Airbnbs throughout the New York City area in 2019. There are 48895 different Airbnb's presented in this data set. 

**Data Attribute Summary:**

|Name of attribute|Type of data|Semantics|Notes|
|---|---|---|---|
|`id`|Quantitative `dbl`|There is a unique value in each of these columns, id is essentially the index for the row|Unique value for each row, does not start at `0`.|
|`name`|Categorical variable `chr`|This is the name of the BnB that the row is about.|There are `47895` unique values in this column. Meaning it is not unique throughout the whole data set.|
|`host_id`|Quantitative variable `dbl`|This is the numerical ID for the BnB host for the row|`37457` different values in this column ranging from `2438` to `274321313`|
|`host_name`|Categorical variable `chr`|Name of the host in charge of the Bnb for the row|`11453` unique values in this row. The names are mainly just the first name of the Bnb host.|
|`neighbourhood_group`|Categorical variable `chr`|The neighbourhood are grouped together in this data. These are general community groupings in the New York area|There are 5 unique values in this column. These values include: `Brooklyn`, `Manhattan`, `Queens`, `Staten Island` and `Bronx`.|
|`neighbourhood`|Categorical variable `chr`|This is the neighbourhood in New York the Airbnb can be found in|There are 221 unique values in this column. They are represented as strings where each word begins with a capital.|
|`latitude`|Quantitative variable `dbl`|This is the latitude location of the BnB represented in each row|Ranges from `40.50` to `40.91`. All within the New York area.|
|`longitude`|Quantitative variable `dbl`|This is the longitude of the Bnb represented in each row|Ranges from`-74.24` to `73.71`|
|`room_type`|Categorical variable `chr`|This is the type of room at you can rent out through each Bnb|Has 3 unique values. These values include: `Private room`, `Entire home/apt`, and `Shared room`.|
|`price`|Quantitative variable `dbl`|This is how much is costs to stay with each Airbnb for a single night|Ranges from `0.0` to `10000.0`. All values are a rounded whole number.|
|`minimum_nights`|Quantitative variable `dbl`|This is the minimum amount of nights someone must book to stay at this bnb.|Ranges from `1.00` to `1250.0`, all values are rounded whole numbers.|
|`number_of_reviews`|Quantitative variable `dbl`|This is the number of reviews this Airbnb recieved prior to this study|Ranges from `0.0` to `629.00`. All values are whole numbers.|
|`last_review`|Date `date`|This is the day that the last review (prior to this study) was posted.|Values range from `2011`, to `2019`. There are also `10052` NA values in this column.|
|`reviews_per_month`|Quantitative variable `dbl`|This is the average number of reviews the Bnb recieves each month|Ranges from `0.010` to `58.5`. With a mean of `1.373` reviews each month.|
|`calculated_host_listings_count`|Quantitative variable `dbl`|This is the number of other Airbnbs this host runs.|Ranges from `1` to `327`. Most of this column is made up of 1's (the host runs only one bnb and it is the bnb represented in that row.|
|`availability_365`|Quantitative variable `dbl`|This is the number of days available for booking (not yet booked when data was collected).|Ranges from `0` to `365`. The average amount of days available is `112`.|

---
## Question:

**Primary research question:**

> Using the location, availability, room type, & number of listings the host has, how much should a single night stay at the Airbnb cost?

This question is aimed towards new Airbnb hosts who are unsure about how much a stay at their bnb should cost. My goal is to build a predictive model that will predict an appropriate price for a nights stay at an Airbnb based on the variables listed above. For location, I will be looking at the `neighbourhood_group` variable as this is often how people would choose a location of an Airbnb. All other location variables are going to be dropped to avoid multi-collinearity.

Additionally, I decided to remove the number of reviews and other review information from this model. Most often, people leave reviews when they have recieved really good services or extremely poor services. Since sentiment was not looked at in this data collection I am dropping this variable as it could nullify the models predictions. 